library/src/main/java/j2html/tags/ContainerTag.java

```diff
@@ -143,100 +147,40 @@ public int getNumChildren() {
      */
     public String renderFormatted() {
         try {
-            return renderFormatted(0);
-        } catch (IOException e) {
+            return render(IndentedHtml.into(new StringBuilder(), Config.global())).toString();
+        }catch (IOException e) {
             throw new RuntimeException(e.getMessage(), e);
         }
     }
 
-    private static void indent(Appendable appendable, int level) throws IOException {
-        appendable.append(Config.indenter.indent(level, ""));
-    }
-
-    private String renderFormatted(int lvl) throws IOException {
-        StringBuilder sb = new StringBuilder();
-        renderFormatted(lvl, sb);
-        return sb.toString();
-    }
-
-
-    private void renderFormatted(int lvl, Appendable sb) throws IOException {
-
-        renderOpenTag(sb, null);
-        if (hasTagName() && !isSelfFormattingTag()) {
-            sb.append("\n");
-        }
-        if (!children.isEmpty()) {
-            for (DomContent c : children) {
-                lvl++;
-                if (c instanceof ContainerTag) {
-                    if (((ContainerTag) c).hasTagName()) {
-                        if (this.isSelfFormattingTag()) {
-                            c.render(sb);
-                        } else {
-                            indent(sb, lvl);
-                            ((ContainerTag<?>) c).renderFormatted(lvl, sb);
-                        }
-                    } else {
-                        if (this.isSelfFormattingTag()) {
-                            c.render(sb);
-                        } else {
-                            ((ContainerTag<?>) c).renderFormatted(lvl-1, sb);
-                        }
-                    }
-                } else {
-                    if (!this.isSelfFormattingTag()) {
-                        indent(sb, lvl);
-                    }
-                    c.render(sb);
-                    if (!this.isSelfFormattingTag()) {
-                        sb.append("\n");
-                    }
-                }
-                lvl--;
-            }
-        }
-        if (hasTagName() && !this.isSelfFormattingTag()) {
-            indent(sb, lvl);
-        }
-        renderCloseTag(sb);
+    @Override
+    public <T extends Appendable> T render(HtmlBuilder<T> builder, Object model) throws IOException {
```

Comment:
There's already a type parameter `T` at the class level, so perhaps this should be
```suggestion
    public <A extends Appendable> A render(HtmlBuilder<A> builder, Object model) throws IOException {
```